libcurv/string.h

```diff
@@ -85,22 +85,38 @@ operator<<(std::ostream& out, Shared<const String> str)
     return out;
 }
 
-/// Make a curv::String from an array of characters
+// Make a curv::String from an array of characters
 Shared<String> make_string(const char* str, size_t len);
 
-/// Make a curv::String from a character Range
+// Make a curv::String from a character Range
 inline Shared<String> make_string(Range<const char*> r)
 {
     return make_string(r.begin(), r.size());
 }
 
-/// Make a curv::String from a C string
+
+// Make a curv::String from a C string
 inline Shared<String>
 make_string(const char*str)
 {
     return make_string(str, strlen(str));
 }
 
+// Deleted overload, use other overloads!
+//
+// If you wish to convert a boost::filesystem::path to Shared<String>,
+// call make_string(path.string().c_str()).
+//
+// If you really have a pure wchar_t* string (e.g. from Windows API),
+// use the conversion function from <libcurv/win32.h> to convert it to
+// a char* string.
+//
+// Rationale for explicit deletion: without deletion, calling
+// make_string(wchar_t*) would use the `make_string(Value);` overload --
+// which is wrong since the implicit cast to Value makes the string "true"
+// (literally "true") out of wchar_t* strings, independent of their contents.
+Shared<String> make_string(const wchar_t*) = delete;
```

Comment:
Even though you said

> so I don't want any changes to libcurv/string.h.

I added this deleted overload. Namely, before adding it `./curv -x 2+2` always told me `ERROR "true": no such file or directory`. It turned out taht in `make_system()`, where paths to Curv's stdlib are created, there had been a `make_string(path.c_str())` call -- which due to implicit conversion and overloading to `make_string(Value)` always resulted in `"true"`. The call now is `make_string(path.string().c_str())` and the deleted overload there to explicitly throw an error when one is using `make_string(path.c_str())`!